apps/studio/components/interfaces/Settings/Logs/LogsFilterPopover.tsx

```diff
@@ -1,94 +1,124 @@
-import { Fragment, useState } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 
-import { Button, Checkbox, Form, Popover } from 'ui'
+import { Button, cn } from 'ui'
 import type { FilterSet, Filters } from './Logs.types'
+import { Popover, PopoverContent, PopoverTrigger } from '@ui/components/shadcn/ui/popover'
+import { Label } from '@ui/components/shadcn/ui/label'
+import { Checkbox } from '@ui/components/shadcn/ui/checkbox'
 
 interface LogsFilterPopoverProps {
   options: FilterSet
   filters: Filters
   onFiltersChange: (filters: Filters) => void
   buttonClassName: string
+  align?: 'start' | 'end' | 'center'
+  isLoading?: boolean
 }
 
 const LogsFilterPopover = ({
   options,
   filters,
   onFiltersChange,
   buttonClassName,
+  align = 'start',
+  isLoading = false,
 }: LogsFilterPopoverProps) => {
+  const filterKey = options.key
   const [open, setOpen] = useState(false)
+  const [values, setValues] = useState<Filters>(filters)
 
   const handleReset = () => {
-    onFiltersChange({})
-    setOpen(false)
+    const emptyFilters = { [filterKey]: {} }
+    setValues(emptyFilters)
+    onFiltersChange({
+      ...filters,
+      [filterKey]: {},
+    })
   }
+
   const handleToggle = () => setOpen(!open)
-  const checkIsActive = () => {
-    const filter = filters[options.key]
+  const isActive = useMemo(() => {
+    const filter = values[filterKey]
     if (typeof filter === 'object' && Object.values(filter).some(Boolean)) {
       return true
     }
     return false
-  }
+  }, [values, filterKey])
+
+  // Reset values on close without applying
+  useEffect(() => {
+    if (!open) {
+      setValues(filters)
+    }
+  }, [open, filters])
 
   return (
-    <Popover
-      className="flex items-center"
-      open={open}
-      align="end"
-      size="medium"
-      onOpenChange={handleToggle}
-      overlay={
-        <Form
-          className="w-full"
-          initialValues={filters}
-          onSubmit={(values, { setSubmitting }) => {
-            setSubmitting(true)
+    <Popover open={open} onOpenChange={handleToggle}>
+      <PopoverTrigger asChild>
+        <Button
+          type={isActive ? 'default' : 'outline'}
+          onClick={handleToggle}
+          className={cn(
+            'min-w-20 border-dashed',
+            {
+              'border-solid': isActive,
+            },
+            buttonClassName
+          )}
+        >
+          <span>{options.label}</span>
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent align={align} className="p-0 w-60">
```

Comment:
Small nit: can we make the width match the pills width?
![Screenshot 2025-02-05 at 08 44 40](https://github.com/user-attachments/assets/0a032a41-74b3-4273-bc0d-8a58df33a1f0)